File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.java
class Solution {
    
    public StringBuilder invert(StringBuilder prev){
        for(int i=0;i<prev.length();i++){
            if(prev.charAt(i) == '0')
                prev.setCharAt(i,'1');
            else
                prev.setCharAt(i,'0');
        }
        prev.reverse();
        return prev;
    }
    
    public char findKthBit(int n, int k) {
        if(n == 1)
            return '0';
        
        StringBuilder prev = new StringBuilder("0");
        StringBuilder curr = null;
        int i = n;
        
        while(i>1){
            curr = new StringBuilder();
            curr.append(prev);
            curr.append('1');
            curr.append(invert(prev));
            prev = curr;  
            i--;
        }
        
        return curr.charAt(k-1);
    }
}

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string invert(string &prev){
        for(int i=0;i<prev.size();i++){
            if(prev[i] == '0')
                prev[i] = '1';
            else
                prev[i] = '0';
        }
        reverse(prev.begin(),prev.end());
        return prev;
    }
    
    char findKthBit(int n, int k) {
        if(n == 1)
            return '0';
        
        string prev = "0";
        string curr  = "";
        int i = n;
        
        while(i>1){
            curr = prev + "1" + invert(prev);
            prev = curr;
            i--;
        }
        return curr[k-1];
    }
};

